server/src/test/java/org/elasticsearch/common/geo/GeometryIndexerTests.java

```diff
@@ -155,32 +157,59 @@ public void testLine() {
      */
     public double length(Line line) {
         double distance = 0;
+        double[] prev = new double[]{line.getLon(0), line.getLat(0)};
+        GeoUtils.normalizePoint(prev, false, true);
         for (int i = 1; i < line.length(); i++) {
-            distance += Math.sqrt((line.getLat(i) - line.getLat(i - 1)) * (line.getLat(i) - line.getLat(i - 1)) +
-                (line.getLon(i) - line.getLon(i - 1)) * (line.getLon(i) - line.getLon(i - 1)));
+            double[] cur = new double[]{line.getLon(i), line.getLat(i)};
+            GeoUtils.normalizePoint(cur, false, true);
+            distance += Math.sqrt((cur[0] - prev[0]) * (cur[0] - prev[0]) + (cur[1] - prev[1]) * (cur[1] - prev[1]));
+            prev = cur;
         }
         return distance;
     }
 
     /**
-     * A simple tests that generates a random lines crossing anti-merdian and checks that the decomposed segments of this line
+     * Removes the points on the antimeridian that are introduced during linestring decomposition
+     */
+    public static MultiPoint remove180s(MultiPoint points) {
+        List<Point> list = new ArrayList<>();
+        points.forEach(point -> {
+            if (Math.abs(point.getLon()) - 180.0 > 0.000001) {
```

Comment:
is there a specific reason why `0.000001` was chosen here. or is it just a "close enough" difference

Reply:
It is a test. So, I didn't spend too much time thinking about how precise I need to go here. So, it is basically "close enough".

Reply:
sgtm